### PR TITLE
Remove Type::ScalarType()

### PR DESCRIPTION
**Pull request by li-roy**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #17530 Small clean up of aten_op&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14237931/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #17601 Store ScalarType and Backend instead of Type in TensorIterator&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14274754/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #17785 Remove Type::elementSizeInBytes&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14379074/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #17723 Store python default type as PyTensorType instead of at::Type&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14351082/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #17786 Pass ScalarType separately from Type in python constructors&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14379075/)
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#17792 Remove Type::ScalarType()**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14381774/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #17603 Remove conversion operator from Type to TensorOptions&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D14276588/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #17787 Add ScalarType arg to Type::options()&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D14379076/)

Differential Revision: [D14381774](https://our.internmc.facebook.com/intern/diff/D14381774/)

### CODE REVIEW

File: aten/src/ATen/templates/TypeDefault.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ void TypeDefault::set_data(Tensor & self, Tensor new_data) const {
 }
 
 Type & TypeDefault::toBackend(Backend b) const {
-  return at::globalContext().getNonVariableType(b,scalarType());
+  return at::globalContext().getNonVariableType(b,typeMetaToScalarType(typeMeta()));
```
Comment (li-roy) [inline aten/src/ATen/templates/TypeDefault.cpp]: This change is orthogonal and not making progress (typeMeta() needs to go), but I'd like to just get rid of scalarType() for now.

File: torch/csrc/tensor/python_tensor.cpp [hunk 2]
```diff
@@ -345,7 +345,7 @@ void py_set_default_dtype(PyObject* obj) {
 
 void set_default_tensor_type(PyTensorType& type) {
   auto aten_type = type.aten_type();
-  if (!at::isFloatingType(aten_type->scalarType())) {
+  if (!at::isFloatingType(static_cast<at::ScalarType>(type.scalar_type))) {
```
Comment (ezyang) [inline torch/csrc/tensor/python_tensor.cpp]: I'm surprised you need a cast here

**Review by ezyang — APPROVED** [review]
whooooooo
